application-admintools-default/src/main/java/com/xwiki/admintools/internal/data/identifiers/CurrentServer.java

```diff
@@ -93,7 +99,9 @@ public void updateCurrentServer()
     {
         this.currentServerInfo = null;
         for (ServerInfo serverInfo : this.supportedServers.get()) {
-            if (serverInfo.isUsed()) {
+            boolean matchingHint = usageDataProvider.getServerMetadata().get(SERVER_NAME_KEY).toLowerCase()
+                .contains(serverInfo.getComponentHint());
+            if (matchingHint && serverInfo.isUsed()) {
```

Comment:
It seems to me that `matchingHint` and `serverInfo.isUsed()` do the same thing. The `UsageDataProvider#getServerMetadata` returns the metadata of used servlet container (tomcat/jetty) and `ServerInfo#isUsed` returns whether that specific server is used or not. Am I mistaken? Or are there specific cases in which `#isUsed` method might return false even tho the server is actually running on Tomcat, for example?

Reply:
The functionality between the two is different. The `matchingHint` checks that the `serverInfo` is indeed checking the path for the used server, and the `serverInfo.isUsed()` checks if a path to the identified server is correct. Indeed, with this addition, the name is confusing. I changed from `serverInfo.isUsed()` to `serverInfo.foundServerPath()`.